### DISCUSSION TITLE
How to Update

### DISCUSSION THREAD

**Discussion by jake-b** [solokeys/solo2-cli#21] [Q&A]
How do you perform a firmware update?

Using the compiled solo2 binary:

```
% ./solo2 update                                    
Downloading latest release from https://github.com/solokeys/solo2/
Fetched firmware version 1:20200101.8
Error: Empty list of Solo 2 devices
```

Can't seem to get it to see the connected SoloKey2.

**Comment by raspberrycoulis** [comment]
Take a look at https://github.com/solokeys/solo2-cli/discussions/19

**Comment by jake-b** [comment]
Thanks, Missed all the detail inside the collapsed thread.  I can get everything to compile but still no go (list is empty).  Possibly because I'm on an ARM Mac.  (Instructions mention that it might need some additional tweaks for this case). I'll take a pass on an Intel Mac.

**Reply by nickray** [reply]
Yeah, Mac is kinda stuck. Eventually its fork of PCSC lite will get our VID:PID pair from upstream.
As mentioned in a separate response to that discussion, working on exposing the CTAP-based update mechanism, which would then work on Mac (and conservative Linux distros..) without, or with out-of-date, PCSC lite.

Windows is the odd one out: You can't use custom CTAP without root, but you have PCSC for free :)
